src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_NAME_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFromFile(fromFileName).split(" ");
+        String report = createReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    public String createReport(String[] dataFromFile) {
+        int buy = 0;
+        int supply = 0;
+        for (int i = 0; i < dataFromFile.length; i++) {
+            String[] value = dataFromFile[i].split(",");
+            if ("supply".equals(value[OPERATION_NAME_INDEX])) {
+                supply += Integer.parseInt(value[VALUE_INDEX]);
+            } else {
+                buy += Integer.parseInt(value[VALUE_INDEX]);
+            }
+        }
+
+        return calculateResult(supply, buy);
+    }
+
+    public String readFromFile(String fileName) {
+        File file = new File(fileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
```

Comment:
Please use try-with-resources as mentioned in common mistakes:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources